File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/swigobject_runme.m
# do not dump Octave core
if exist("crash_dumps_octave_core", "builtin")
  crash_dumps_octave_core(0);
endif

swigobject

a = A();

a1 = a_ptr(a);
a2 = a_ptr(a);

if (swig_this(a1) != swig_this(a2))
  error
endif
  

lthis = uint64(swig_this(a.this));
xstr1 = printf("0x%x",lthis);
xstr2 = pointer_str(a);

if (xstr1 != xstr2)
  error
endif

s = str(a.this);
r = repr(a.this);

v1 = v_ptr(a);
v2 = v_ptr(a);
if (uint64(v1) != uint64(v2))
  error
endif

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/swigobject_runme.py

from swigobject import *

a = A()


a1 = a_ptr(a)
a2 = a_ptr(a)

if a1.this != a2.this:
    raise RuntimeError


lthis = long(a.this)
# match pointer value, but deal with leading zeros on 8/16 bit systems and
# different C++ compilers interpretation of %p
xstr1 = "%016X" % (lthis,)
xstr1 = str.lstrip(xstr1, '0')
xstr2 = pointer_str(a)
xstr2 = str.replace(xstr2, "0x", "")
xstr2 = str.replace(xstr2, "0X", "")
xstr2 = str.lstrip(xstr2, '0')
xstr2 = str.upper(xstr2)

if xstr1 != xstr2:
    print xstr1, xstr2
    raise RuntimeError

s = str(a.this)
r = repr(a.this)

v1 = v_ptr(a)
v2 = v_ptr(a)
if long(v1) != long(v2):
    raise RuntimeError
